test/jdk/java/foreign/TestLayoutPaths.java

```diff
@@ -343,22 +396,76 @@ public void testSequencePaths() {
 
         // test select
 
-        MemoryLayout selected = g.select(PathElement.sequenceElement());
+        MemoryLayout selected = g.select(sequenceElement());
         assertTrue(selected == MemoryLayouts.JAVA_BYTE);
 
         // test offset
 
         for (int i = 0 ; i < 4 ; i++) {
-            long bitOffset = g.bitOffset(PathElement.sequenceElement(i));
+            long bitOffset = g.bitOffset(sequenceElement(i));
             assertEquals(offsets[i], bitOffset);
-            long byteOffset = g.byteOffset(PathElement.sequenceElement(i));
+            long byteOffset = g.byteOffset(sequenceElement(i));
             assertEquals((offsets[i]) >>> 3, byteOffset);
         }
 
         // test map
 
-        SequenceLayout seq2 = (SequenceLayout)g.map(l -> MemoryLayouts.JAVA_DOUBLE, PathElement.sequenceElement());
+        SequenceLayout seq2 = (SequenceLayout)g.map(l -> MemoryLayouts.JAVA_DOUBLE, sequenceElement());
         assertTrue(seq2.elementLayout() == MemoryLayouts.JAVA_DOUBLE);
     }
+
+    @Test(dataProvider =  "offsetHandleCases")
+    public void testOffsetHandle(MemoryLayout layout, PathElement[] pathElements, long[] indexes,
+                                 long expectedBitOffset) throws Throwable {
+        MethodHandle bitOffsetHandle = layout.bitOffsetHandle(pathElements);
+        bitOffsetHandle = bitOffsetHandle.asSpreader(long[].class, indexes.length);
+        long actualBitOffset = (long) bitOffsetHandle.invokeExact(indexes);
+        assertEquals(actualBitOffset, expectedBitOffset);
+        if (expectedBitOffset % 8 == 0) {
+            MethodHandle byteOffsetHandle = layout.byteOffsetHandle(pathElements);
+            byteOffsetHandle = byteOffsetHandle.asSpreader(long[].class, indexes.length);
+            long actualByteOffset = (long) byteOffsetHandle.invokeExact(indexes);
+            assertEquals(actualByteOffset, expectedBitOffset / 8);
+        }
+    }
+
+    @DataProvider
+    public static Object[][] offsetHandleCases() {
+        return new Object[][]{
+            {
+                MemoryLayout.ofSequence(10, JAVA_INT),
```

Comment:
The test is good, but there's no case which features more than one long index. I think it is worth adding at least a few, since that's where all the combination logic (and ordering) can go wrong?

Reply:
Yes, makes sense. Thanks for the suggestion.